### DISCUSSION TITLE
question on nc-input demo

### DISCUSSION THREAD

**Discussion by kascote** [dankamongmen/notcurses#2511] [Q&A] [answered]
Hi, I'm toying doing a binding for `dart` and was trying to replicate the nc-input demo to test some implementation details.
Following the nc-input code I saw something that is not clear to me (I do not understand some things of c,c++ and I'm new to notcurses too).

this line is printing some text using`NCAlign::Center`
https://github.com/dankamongmen/notcurses/blob/master/src/input/input.cpp#L292

but the `n` plane was not created with `HORALIGNED` flag
https://github.com/dankamongmen/notcurses/blob/master/src/input/input.cpp#L258

`HORALIGNED` is only setup for the "plot" plane
https://github.com/dankamongmen/notcurses/blob/master/src/input/input.cpp#L274

The question is: why the text is displayed centered when run the demo if the plane do not has that flag setup?

**Comment by dankamongmen** [comment, marked as answer]
`NCPLANE_HORALIGNED` isn't about the text within the plane, but rather about the plane's placement relative to its parent plane.

when i create a plane with `NCPLANE_HORALIGNED` using `NCALIGN_CENTER`, i'm saying "stick this plane horizontally in the middle of its parent plane." so if say the standard plane is 80 columns wide, and i make a new child plane using `NCALIGN_CENTER` and the new plane is 40 columns wide, it'll be placed at the 20th column.

within a plane, using `NCALIGN_CENTER` together with e.g. `ncplane_printf_aligned()` aligns *within the plane*.

does that answer your question?

**Reply by kascote** [reply]
yes perfect.. `putstr` is overloaded a couple times, I was looking at the wrong one. 
thanks an have Happy New Year!
